sling.go

```diff
@@ -145,6 +146,20 @@ func (s *Sling) Set(key, value string) *Sling {
 	return s
 }
 
+// SetBasicAuth mimics http.Request.SetBasicAuth()
```

Comment:
// SetBasicAuth sets the Authorization header to use HTTP Basic Authentication
// with the provided username and password. With HTTP Basic Authentication
// the provided username and password are not encrypted.

^ let's match net/http so its implicitly clear it does the same.